src/flamegpu/simulation/CUDAEnsemble.cu

```diff
@@ -199,11 +211,274 @@ unsigned int CUDAEnsemble::simulate(const RunPlanVector &plans) {
         step_log_config.get(), exit_log_config.get(), step_log_config && step_log_config->log_timing, exit_log_config && exit_log_config->log_timing);
     }
 
-    // Wait for all runners to exit
-    for (unsigned int i = 0; i < TOTAL_RUNNERS; ++i) {
-        runners[i].thread.join();
-        runners[i].~SimRunner();
+    // In OMP mode, only Rank 0 increments the error counter
+    unsigned int err_count = 0;
+    if (config.mpi) {
+#ifdef FLAMEGPU_ENABLE_MPI
```

Comment:
This ordering means that the mpi can be enabeld in the config, but not compiled in? so there's no guarantee that somethign will happen? 

I assume its this way round so the else clause works without needing to be duplicated though?

Reply:
`config.mpi` is `const false` when `FLAMEGPU_ENABLE_MPI` is not set. So the branch will be ignored by the compiler.